HOMEWORK.js

```diff
@@ -58,31 +58,36 @@ function homework (renderList) {
 	// ◽ super bonus: remove the X if it already has one (put it back to incomplete)
 	function onCompleteTodo (index) {
 		// Add code here
-
+		todos[index].charAt(0) === 'X' ? todos[index] = todos[index].slice(1) : todos[index] = 'X' + todos[index];
 		renderList(todos);
 	}
 
 	// ✅This function should:
 	// ◽ remove from the list all of the completed items (that start with 'X')
 	function onRemoveCompleted () {
 		// Add code here
-
+		todos.filter(item => item.charAt(0) === 'X').map(item => todos.splice(todos.indexOf(item),1));
 		renderList(todos);
 	}
 
 	// ✅This function should:
 	// ◽ add an exclamation mark ('!') to the end of every item in the list
 	function onUrgent () {
 		// Add code here
-		
+		todos.forEach(item => todos[todos.indexOf(item)] = item + "!");
 		renderList(todos);
 	}
 
 	// ✅This function should:
 	// ◽ display an alert with the first item on the list not marked as done (first without an 'X')
 	// ◽ bonus: consider what will happen if there are no more items left to do
 	function onWhatNext () {
-		alert('This is just a dummy message!');
+		var next = todos.find(item => item.charAt(0) !== "X");
```

Comment:
Great use of `find`.

This is the third time you've checked whether a todo starts with `X` though. What could you add to the code so that you don't need to repeat yourself in this way?